### PR TITLE
Add CORS to backend for the new frontend setup

### PR DESCRIPTION
**Pull request by harsh3dev**
This pull request introduces support for Cross-Origin Resource Sharing (CORS) in the backend configuration. The key changes include adding the django-cors-headers package, updating the installed apps, middleware, and settings to enable CORS.

Changes to support CORS:
- Used django-cors-headers to allow CORS headers.
- Configured CORS_ALLOWED_ORIGINS to allow requests from http://localhost:3000 and http://localhost:8000.

'*This will be beneficial for implementing the various pages in the new frontend setup.*

### CODE REVIEW

File: backend/settings/base.py [hunk 1]
```diff
@@ -49,6 +50,8 @@ class Base(Configuration):
     INSTALLED_APPS = DJANGO_APPS + THIRD_PARTY_APPS + LOCAL_APPS
 
     MIDDLEWARE = [
+        "corsheaders.middleware.CorsMiddleware",
```
Comment (arkid15r) [inline backend/settings/base.py:53]: I know this must go first but I believe `django.middleware.common.CommonMiddleware` can stay where it was

File: backend/settings/base.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ class Base(Configuration):
     SITE_URL = "http://localhost:8000"
 
     DJANGO_APPS = (
+        "corsheaders",
```
Comment (arkid15r) [inline backend/settings/base.py]: Move it to `THIRD_PARTY_APPS` in alphabetical order please

File: backend/settings/base.py [hunk 3]
```diff
@@ -58,6 +61,11 @@ class Base(Configuration):
         "django.middleware.clickjacking.XFrameOptionsMiddleware",
     ]
 
+    CORS_ALLOWED_ORIGINS = [
```
Comment (arkid15r) [inline backend/settings/base.py]: Let's keep the defaults empty and set values for each env in corresponding files, e.g. CORS_ALLOWED_ORIGINS = ("http://localhost:3000", "http://127.0.0.1:3000") for `local.py`

File: frontend/src/pages/Projects.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { useEffect } from "react"
+
+
+export default function Projects() {
+
+    useEffect(()=>{
+        document.title = "OWASP Projects"
+        const fetchApiData = async () => {
+            try {
+                const response = await fetch(`http://localhost:8000/api/v1/owasp/search/project?q=`)
```
Comment (arkid15r) [inline frontend/src/pages/Projects.tsx]: It looks like it's time for frontend env variables introduction. We need an BACKEND_URL / API_URL value for the new pages. Let's focus on that in a separate PR.

**Review by arkid15r — COMMENTED** [review]
Thanks for adding this! Please look into my comments when you got a chance:

File: frontend/src/pages/Projects.tsx (hunk 4 above)
Comment (harsh3dev) [reply frontend/src/pages/Projects.tsx]: I will set those env for frontend in the next PR

File: backend/settings/local.py [hunk 5]
```diff
@@ -16,3 +16,7 @@ class Local(Base):
     APP_NAME = "OWASP Nest Local"
 
     LOCAL_EXTERNAL_IP = values.SecretValue(environ_name="LOCAL_EXTERNAL_IP")
+    CORS_ALLOWED_ORIGINS = [
+        "http://localhost:3000",
+        "http://localhost:8000",
+    ]
```
Comment (arkid15r) [inline backend/settings/local.py]:
```suggestion
    CORS_ALLOWED_ORIGINS = (
        "http://localhost:3000",
        "http://localhost:8000",
    )
```

**Review by arkid15r — APPROVED** [review]
LGTM

**Review by arkid15r — APPROVED** [review]
Merging this, thanks @harsh3dev !

**Pull request merged**